### PR TITLE
Drop redundant dev code part 3

### PR DESCRIPTION
**Pull request by roman-khimov**
Moving forward, see #307, #315 and #318.

### CODE REVIEW

**Comment by im-kulikov** [comment]
Not sure, that it is good idea to read or write after the errors..

We should return errors without continuing code execution

**Comment by roman-khimov** [comment]
> **Propose make constructor for BinReader / BinWriter, for example:**

New issue, please. Here, we're mostly about merging code from `_pkg.dev` and not rewriting it at the same time.

> Not sure, that it is good idea to read or write after the errors..

I think this one is addressed already with 7b46f9bb860853a2fb44518c260f745c5e012667.

**Comment by im-kulikov** [comment]
> I think this one is addressed already with 7b46f9b.

nope.. 

https://github.com/nspcc-dev/neo-go/blob/4f23117d32c87cd5c5d6b7e3906ce83902e6e76f/pkg/util/binaryWriter.go#L69-L78

https://github.com/nspcc-dev/neo-go/blob/4f23117d32c87cd5c5d6b7e3906ce83902e6e76f/pkg/util/binaryReader.go#L35-L75

**Comment by roman-khimov** [comment]
>>    I think this one is addressed already with 7b46f9b.
> nope..

These are left out intentionally, in case of failed reads they have safe behavior.

**Comment by im-kulikov** [comment]
https://github.com/nspcc-dev/neo-go/blob/4f23117d32c87cd5c5d6b7e3906ce83902e6e76f/pkg/util/binaryReader.go#L43-L57

```go
	switch b {
	case 0xfd:
		var v uint16
		r.Err = binary.Read(r.R, binary.LittleEndian, &v)
		return uint64(v)
	case 0xfe:
		var v uint32
		r.Err = binary.Read(r.R, binary.LittleEndian, &v)
		return uint64(v)
	case 0xff:
		var v uint64
		r.Err = binary.Read(r.R, binary.LittleEndian, &v)
		return v
	default:
		return uint64(b)
	}
```

File: pkg/util/binaryReader.go [hunk 1]
```diff
@@ -12,27 +12,31 @@ type BinReader struct {
...
+// into the interface v in big-endian format
+func (r *BinReader) ReadBE(v interface{}) {
 	if r.Err != nil {
 		return
 	}
 	r.Err = binary.Read(r.R, binary.BigEndian, v)
 }
 
-//VarUint reads a variable integer from the
+// ReadVarUint reads a variable-length-encoded integer from the
```
Comment (volekerb) [inline pkg/util/binaryReader.go]: nit: probably I would add smth like:
"in case of error returns 0"

**Pull request merged**